### PR TITLE
JDK-8304689: Add hidden option to disable external spec page

### PR DESCRIPTION
**Pull request by jonathan-gibbons**
Please review a change to introduce a hidden option to temporarily disable the "External Specifications" page while we add `@spec` tags, and until we have a critical mass of such tags.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8304689](https://bugs.openjdk.org/browse/JDK-8304689): Add hidden option to disable external spec page

### Reviewers
 * [Erik Joelsson](https://openjdk.org/census#erikj) (@erikj79 - **Reviewer**) ⚠️ Review applies to [d89e76a0](https://git.openjdk.org/jdk/pull/13127/files/d89e76a0526d61eca548ddcdc90c7d9ea4228001)
 * [Hannes Wallnöfer](https://openjdk.org/census#hannesw) (@hns - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/13127/head:pull/13127` \
`$ git checkout pull/13127`

Update a local copy of the PR: \
`$ git checkout pull/13127` \
`$ git pull https://git.openjdk.org/jdk.git pull/13127/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 13127`

View PR using the GUI difftool: \
`$ git pr show -t 13127`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/13127.diff">https://git.openjdk.org/jdk/pull/13127.diff</a>

</details>

### CODE REVIEW

**Review by erikj79 — APPROVED** [review]
Build change looks ok.

File: src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/HtmlOptions.java [hunk 1]
```diff
@@ -343,6 +350,14 @@ public boolean process(String opt,  List<String> args) {
                     }
                 },
 
+                new Hidden(resources, "--no-external-specs-page") {
+                    @Override
+                    public boolean process(String opt,  List<String> args) {
```
Comment (turbanoff) [inline src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/HtmlOptions.java:355]: nit
```suggestion
                    public boolean process(String opt, List<String> args) {
```

**Review by hns — APPROVED** [review]
Looks good to me!

**Pull request closed without merging**